packages/react/docs/guides/tech-stack.md

```diff
@@ -0,0 +1,283 @@
+
+There are crucial differences between arrow functions and traditional functional expressions so be aware and know [when to not use an arrow function](https://www.javascripttutorial.net/es6/when-you-should-not-use-arrow-functions/). With the above information in mind, I would lean towards using function delcaration for most instances and arrow functions when used as an argument in another function.
+
+When writing a functional component it is best practice to use a regular function declaration
+
+```js
+function MyComponent(props) {
+  return <div>Hello World</div>;
+}
+```
+
+but you will definitely see them defined using arrow functions both in our library and in the wild.
+
+```js
+const MyComponent = (props) => <div>Hello World</div>;
+```
+
+## [Template Literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
+
+are literals delimited with backticks (`), allowing embedded expressions called substitutions.
+
+```js
+const myString = `This is a string with ${variables}.`;
+```
+
+It can make strings with variable or expression iterpolation more readable and easier to write. Notice the `\n` is not needed because white space is not folded into one space as with string literals.
+
+```js
+const str1 = 'Fifteen is ' + (a + b) + ' and\nnot ' + (2 * a + b) + '.';
+const str2 = `Fifteen is ${a + b} and
+not ${2 * a + b}.`;
+```
+
+## [Tertiary conditional](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
+
+is the only JavaScript operator that takes three operands: a condition followed by a question mark (?), then an expression to execute if the condition is truthy followed by a colon (:), and finally the expression to execute if the condition is falsy. This operator is frequently used as an alternative to an if...else statement.
+
+In react there will be times when you want to conditionally render an element. Like the content of an accordion.
+
+```js
+<AccordionItem>{isOpen ? <AccordionItemContent>Hey</AccordionItemContent> : null}</AccordionItem>
+```
+
+## [Optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) (?)
+
+enables you to read the value of a property located deep within a chain of connected objects without having to check that each reference in the chain is valid. The following would normally throw an error
+
+```js
+const adventurer = {
+  name: 'Alice',
+  cat: {
+    name: 'Dinah',
+  },
+};
+
+const dogName = adventurer.dog.name;
+```
+
+and to prevent you would have to check for the existence of each property before using it. Optional chaining allows us to do that in place
+
+```js
+const dogName = adventurer.dog?.name;
+adventurer.dog?.someNonExistentMethod?.();
+
+// vs
+
+if (adventurer.dog && adventurer.dog.someNonExistentMethod) {
+  const dogName = adventurer.dog.name;
+  adventurer.dog.someNonExistentMethod.();
```

Comment:
At 187 line , do we need .() in the end?

Reply:
Do not need the `.`. Thanks